services/horizon/cmd/root.go

```
@@ -40,17 +46,50 @@ func validateBothOrNeither(option1, option2 string) {
 	}
 }
 
+func pingDB(db *sql.DB) {
+	for attempt := 0; attempt < maxDBPingAttempts; attempt++ {
+		err := db.Ping()
+		if err == nil {
+			return
+		}
+		stdLog.Printf("could not ping horizon db: %v\n", err)
```

I imagine that this will make users ask us about this error. I think when a DB is starting and we ping fails we should just ignore the error.